File: SwissO/SQLiteHelper.cs
namespace SwissO {
    interface SQLiteHelper {
        public const string DATABASE_NAME = "SwissO.dat";
        public const int DATABASE_VERSION = 3;

        //Tables
        public const string TABLE_Profil = "Profil";
        public const string TABLE_Freunde = "Freunde";
        public const string TABLE_Laeufer = "Laeufer";
        public const string TABLE_Clubs = "Clubs";
        public const string TABLE_Events = "Events";

        //Primary Key
        public const string COLUMN_ID = "_id";

        //Fremdschlüssel
        public const string COLUMN_Profil = "profil";
        public const string COLUMN_Event = "event";

        //Table Profil
        public const string COLUMN_Vorname = "first_name";
        public const string COLUMN_Nachname = "surname";
        public const string COLUMN_Name = "name";
        public const string COLUMN_SI = "sicard";
        public const string COLUMN_Category = "category";

        //Table Events
        public const string COLUMN_Title = "title";
        public const string COLUMN_SOLVId = "solvid";
        public const string COLUMN_Date = "date";
        public const string COLUMN_Deadline = "deadline";
        public const string COLUMN_Region = "region";
        public const string COLUMN_IntKoordN = "skoordn";
        public const string COLUMN_IntKoordE = "skoorde";
        public const string COLUMN_Club = "club";
        public const string COLUMN_Map = "map";
        public const string COLUMN_LAusschreibung = "ausschreibung";
        public const string COLUMN_LWeisungen = "weisungen";
        public const string COLUMN_LRangliste = "rangliste";
        public const string COLUMN_LLiveRangliste = "liverangliste";
        public const string COLUMN_LStartliste = "startliste";
        public const string COLUMN_LAnmeldung = "anmeldung";
        public const string COLUMN_LMutation = "mutation";
        public const string COLUMN_EntryPortal = "entryportal";

        //Table Laeufer
        public const string COLUMN_Jahrgang = "jahrgang";
        public const string COLUMN_Startnummer = "startnummer";
        public const string COLUMN_Startzeit = "startzeit";
        public const string COLUMN_Zielzeit = "zielzeit";
        public const string COLUMN_Rang = "rang";


        protected const string SQL_Profil = "CREATE TABLE IF NOT EXISTS " + TABLE_Profil + "(" +
                COLUMN_ID + " INTEGER PRIMARY KEY AUTOINCREMENT," +
                COLUMN_Vorname + " VARCHAR(15)," +
                COLUMN_Nachname + " VARCHAR(15)," +
                COLUMN_SI + " INTEGER," +
                COLUMN_Category + " VARCHAR(15))";

        protected const string SQL_Laeufer = "CREATE TABLE IF NOT EXISTS " + TABLE_Laeufer + "(" +
                COLUMN_ID + " INTEGER PRIMARY KEY AUTOINCREMENT," +
                COLUMN_Name + " VARCHAR(31) NOT NULL," +
                COLUMN_Jahrgang + " INTEGER," +
                COLUMN_Club + " VARCHAR(31)," +
                COLUMN_Category + " VARCHAR(15) NOT NULL," +
                COLUMN_Startnummer + " INTEGER," +
                COLUMN_Startzeit + " INTEGER," +
                COLUMN_Zielzeit + " INTEGER," +
                COLUMN_Rang + " INTEGER," +
                COLUMN_Event + " INTEGER NOT NULL)";

        protected const string SQL_Freunde = "CREATE TABLE IF NOT EXISTS " + TABLE_Freunde + "(" +
                COLUMN_ID + " INTEGER PRIMARY KEY AUTOINCREMENT," +
                COLUMN_Name + " VARCHAR(31) NOT NULL," +
                COLUMN_Profil + " INTEGER NOT NULL)";

        protected const string SQL_Clubs = "CREATE TABLE IF NOT EXISTS " + TABLE_Clubs + "(" +
                COLUMN_ID + " INTEGER PRIMARY KEY AUTOINCREMENT," +
                COLUMN_Name + " VARCHAR(31) NOT NULL," +
                COLUMN_Profil + " INTEGER NOT NULL)";

        protected const string SQL_Events = "CREATE TABLE IF NOT EXISTS " + TABLE_Events + "(" +
                COLUMN_ID + " INTEGER PRIMARY KEY AUTOINCREMENT," +
                COLUMN_Title + " VARCHAR(15) NOT NULL," +
                COLUMN_Date + " INTEGER NOT NULL," +
                COLUMN_Region + " VARCHAR(15)," +
                COLUMN_Club + " VARCHAR(31)," +
                COLUMN_Map + " VARCHAR(31)," +
                COLUMN_Deadline + " INTEGER," +
                COLUMN_IntKoordN + " DOUBLE NOT NULL," +
                COLUMN_IntKoordE + " DOUBLE NOT NULL," +
                COLUMN_LAusschreibung + " VARCHAR(255)," +
                COLUMN_LWeisungen + " VARCHAR(255)," +
                COLUMN_LRangliste + " VARCHAR(255)," +
                COLUMN_LLiveRangliste + " VARCHAR(255)," +
                COLUMN_LStartliste + " VARCHAR(255)," +
                COLUMN_LAnmeldung + " VARCHAR(255)," +
                COLUMN_LMutation + " VARCHAR(255)," +
                COLUMN_EntryPortal + " INTEGER NOT NULL)";
    }
}

File: app/src/main/java/ch/laasch/swisso/SQLiteHelper.java
package ch.laasch.swisso;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;

public class SQLiteHelper extends SQLiteOpenHelper {

    public static final String DATABASE_NAME = "SwissO.db";
    public static final int DATABASE_VERSION = 1;

    //Tables
//    public static final String TABLE_Profil = "Profil";
    public static final String TABLE_Freunde = "Freunde";
    public static final String TABLE_Laeufer = "Laeufer";
    public static final String TABLE_Clubs = "Clubs";
    public static final String TABLE_Events = "Events";

    //Primary Key
    public static final String COLUMN_ID = "id";
    public static final String COLUMN_AUTO_ID = "_id";

    //Table Events
    public static final String COLUMN_NAME = "name";
    public static final String COLUMN_BEGIN_DATE = "begin_date";
    public static final String COLUMN_END_DATE = "end_date";
    public static final String COLUMN_DEADLINE = "deadline";
    public static final String COLUMN_KIND = "kind";
    public static final String COLUMN_NIGHT = "night";
    public static final String COLUMN_NATIONAL = "national";
    public static final String COLUMN_REGION = "region";
    public static final String COLUMN_CLUB = "club";
    public static final String COLUMN_MAP = "map";
    public static final String COLUMN_LOCATION = "location";
    public static final String COLUMN_INT_NORD = "int_nord";
    public static final String COLUMN_INT_EAST = "int_east";
    public static final String COLUMN_ENTRYPORTAL = "entryportal";
    public static final String COLUMN_AUSSCHREIBUNG = "ausschreibung";
    public static final String COLUMN_LIVE_RESULTATE = "live_resultate";
    public static final String COLUMN_WEISUNGEN = "weisungen";
    public static final String COLUMN_ANMELDUNG = "anmeldung";
    public static final String COLUMN_RANGLISTE = "rangliste";
    public static final String COLUMN_STARTLISTE = "startliste";
    public static final String COLUMN_TEILNEHMERLISTE = "teilnehmerliste";
    public static final String COLUMN_MUTATION = "mutation";

    //Table Laeufer
    public static final String COLUMN_JAHRGANG = "jahrgang";
    public static final String COLUMN_KATEGORIE = "kategorie";
    public static final String COLUMN_STARTNUMMER = "startnummer";
    public static final String COLUMN_STARTZEIT = "startzeit";
    public static final String COLUMN_ZIELZEIT = "zielzeit";
    public static final String COLUMN_RANG = "rang";
    public static final String COLUMN_EVENT = "event";

    //Table Profil
//    public static final String COLUMN_Vorname = "first_name";
//    public static final String COLUMN_Nachname = "surname";
//    public static final String COLUMN_SI = "sicard";


    /*private static final String SQL_Profil = "CREATE TABLE IF NOT EXISTS " + TABLE_Profil + "(" +
            COLUMN_ID + " INTEGER PRIMARY KEY AUTOINCREMENT," +
            COLUMN_Vorname + " VARCHAR(15)," +
            COLUMN_Nachname + " VARCHAR(15)," +
            COLUMN_SI + " INTEGER," +
            COLUMN_KATEGORIE + " VARCHAR(15))";*/

    private static final String SQL_Freunde = "CREATE TABLE IF NOT EXISTS " + TABLE_Freunde + "(" +
            COLUMN_AUTO_ID + " INTEGER PRIMARY KEY AUTOINCREMENT," +
            COLUMN_NAME + " VARCHAR(31) NOT NULL)";

    private static final String SQL_Clubs = "CREATE TABLE IF NOT EXISTS " + TABLE_Clubs + "(" +
            COLUMN_AUTO_ID + " INTEGER PRIMARY KEY AUTOINCREMENT," +
            COLUMN_NAME + " VARCHAR(31) NOT NULL)";

    private static final String SQL_Events = "CREATE TABLE IF NOT EXISTS " + TABLE_Events + "(" +
            COLUMN_ID + " INTEGER PRIMARY KEY," +
            COLUMN_NAME + " TEXT," +
            COLUMN_BEGIN_DATE + " INTEGER," +
            COLUMN_END_DATE + " INTEGER," +
            COLUMN_DEADLINE + " INTEGER," +
            COLUMN_KIND + " INTEGER," +
            COLUMN_NIGHT + " INTEGER," +
            COLUMN_NATIONAL + " INTEGER," +
            COLUMN_REGION + " TEXT," +
            COLUMN_CLUB + " TEXT," +
            COLUMN_MAP + " TEXT," +
            COLUMN_LOCATION + " TEXT," +
            COLUMN_INT_NORD + " REAL," +
            COLUMN_INT_EAST + " REAL," +
            COLUMN_ENTRYPORTAL + " INTEGER," +
            COLUMN_AUSSCHREIBUNG + " TEXT," +
            COLUMN_LIVE_RESULTATE + " TEXT," +
            COLUMN_WEISUNGEN + " TEXT," +
            COLUMN_ANMELDUNG + " TEXT," +
            COLUMN_RANGLISTE + " TEXT," +
            COLUMN_STARTLISTE + " TEXT," +
            COLUMN_TEILNEHMERLISTE + " TEXT," +
            COLUMN_MUTATION + " TEXT)";

    private static final String SQL_Laeufer = "CREATE TABLE IF NOT EXISTS " + TABLE_Laeufer + " ("
            + COLUMN_ID + " INTEGER PRIMARY KEY,"
            + COLUMN_NAME + " VARCHAR(31) NOT NULL,"
            + COLUMN_JAHRGANG + " INTEGER,"
            + COLUMN_CLUB + " VARCHAR(31),"
            + COLUMN_KATEGORIE + " VARCHAR(15) NOT NULL,"
            + COLUMN_STARTNUMMER + " INTEGER,"
            + COLUMN_STARTZEIT + " INTEGER,"
            + COLUMN_ZIELZEIT + " INTEGER,"
            + COLUMN_RANG + " INTEGER,"
            + COLUMN_EVENT + " INTEGER NOT NULL,"
            + COLUMN_STARTLISTE + " INTEGER,"
            + COLUMN_RANGLISTE + " INTEGER)";


    public SQLiteHelper(Context context) {
        super(context, DATABASE_NAME, null, DATABASE_VERSION);
    }

    public void onCreate(SQLiteDatabase db) {
//        db.execSQL(SQLiteHelper.SQL_Profil);
        db.execSQL(SQLiteHelper.SQL_Freunde);
        db.execSQL(SQLiteHelper.SQL_Clubs);
        db.execSQL(SQLiteHelper.SQL_Events);
        db.execSQL(SQLiteHelper.SQL_Laeufer);
    }

    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
//        if (oldVersion < 3) {
//            db.ExecSQL("DROP TABLE IF EXISTS " + SQLiteHelper.TABLE_Profil);
//            db.ExecSQL("DROP TABLE IF EXISTS " + SQLiteHelper.TABLE_Laeufer);
//        }
//        onCreate(db);
    }
}
